apps/src/code-studio/pd/regional_partner_mini_contact/RegionalPartnerMiniContact.jsx

```diff
@@ -61,7 +83,10 @@ export class RegionalPartnerMiniContact extends React.Component {
       email: this.state.email,
       zip: this.state.zip,
       notes: this.state.notes,
-      source: this.props.sourcePageId
+      source: this.props.sourcePageId,
+      // set null or empty values to undefined so they are ignored
+      role: this.state.role || undefined,
+      grade_levels: this.state.grade_levels || undefined
```

Comment:
`name` and `notes` are not required fields either, but we're not coercing to `undefined` in those cases.  Is this behavior redundant (e.g. is any falsy value ignored?) or should we retroactively apply this behavior to those other non-required fields?

Reply:
Might be worth adding a unit test for this behavior too.

Reply:
Only undefined is ignored. For name and notes, if nothing was ever put in the boxes undefined will be sent and then ignored, which is why I added this behavior. If something was put in the boxes and removed then an empty string would be sent. I think it makes sense to send nothing in this case too, but am open to suggestions. For role and grade levels, without this change we would always send an empty string if they don't click a role. For grade levels it would either send undefined if they never checked anything (and therefore be ignored) or it would send null if they checked and then unchecked boxes, which would not be ignored.

Reply:
:+1: Thanks for explaining! I didn't realize on first read that the different field types would impact the behavior here.

Reply:
Yep! I will add the same coercion for name and notes if there are no objections just so everything is consistent